server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java

```diff
@@ -173,6 +216,38 @@ public TypeParser(BiFunction<String, ParserContext, Builder> builderFunction) {
         @Override
         public Mapper.Builder<?> parse(String name, Map<String, Object> node, ParserContext parserContext) throws MapperParsingException {
             Builder builder = builderFunction.apply(name, parserContext);
+            for (Iterator<Map.Entry<String, Object>> iterator = node.entrySet().iterator(); iterator.hasNext();) {
+                Map.Entry<String, Object> entry = iterator.next();
+                String propName = entry.getKey();
+                Object propNode = entry.getValue();
+                checkNull(propName, propNode);
+                switch (propName) {
+                    case NORMALIZER: // allow lowercase, and custom normalizer
+                        builder.setNormalizer(XContentMapValues.nodeStringValue(propNode));
+                        iterator.remove();
+                        break;
+                    case DOC_VALUES: // allow to set docValues to be false
+                        builder.hasDocValues = XContentMapValues.nodeBooleanValue(propNode);
+                        iterator.remove();
+                        break;
+                    case IGNORE_ABOVE:// allow to set if the length of a field is go above certain limit then ignore the document.
+                        builder.ignoreAbove = XContentMapValues.nodeIntegerValue(propNode);
+                        iterator.remove();
+                        break;
+                    case NULL_VALUE:// allow to set nullValue to others
+                        builder.nullValue = XContentMapValues.nodeStringValue(propNode);
```

Comment:
`NA` is just a normal string, can you describe more details? very thank you.

Reply:
`null_value` is not support in the pr, as we have defined the `value` must be an object. it will be more complicated than we thought. If we want to support `null_value`, I will open a new pr to support.